src/expressions/block-expr.md

```diff
@@ -80,6 +80,64 @@ fn move_by_block_expression() {
 }
 ```
 
+## `async` blocks
+
+> **<sup>Syntax</sup>**\
+> _AsyncBlockExpression_ :\
+> &nbsp;&nbsp; `async` `move`<sup>?</sup> _BlockExpression_
+
+An *async block* is a variant of a block expression which evalutes to
+a *future*. The final expression of the block (if present) determines
+the result value of the future.
+
+Executing an async block is thus similar to executing a closure
+expression: its immediate effect is to produce and return a struct.
```

Comment:
Closures are not described in this way in the reference -- it would be sufficient to say "anonymous type" here.